deploy-cloudrun/src/cloudRun.ts

```diff
@@ -0,0 +1,258 @@
+   *
+   * @returns a Cloud Run service.
+   */
+  async getService(service: Service): Promise<run_v1.Schema$Service> {
+    const authClient = await this.getAuthClient();
+    const getRequest: run_v1.Params$Resource$Projects$Locations$Services$Get = {
+      name: this.getResource(service),
+      auth: authClient,
+    };
+    const serviceResponse: GaxiosResponse<run_v1.Schema$Service> = await this.run.projects.locations.services.get(
+      getRequest,
+      this.methodOptions,
+    );
+    return serviceResponse.data;
+  }
+
+  /**
+   * Retrieves a list of Cloud Run services.
+   *
+   * @returns list of Cloud Run services.
+   */
+  async listServices(): Promise<string[]> {
+    const authClient = await this.getAuthClient();
+    const listRequest: run_v1.Params$Resource$Projects$Locations$Services$List = {
+      parent: this.parent,
+      auth: authClient,
+    };
+    const serviceListResponse: GaxiosResponse<run_v1.Schema$ListServicesResponse> = await this.run.projects.locations.services.list(
+      listRequest,
+      this.methodOptions,
+    );
+    const serviceList: run_v1.Schema$ListServicesResponse =
+      serviceListResponse.data;
+    let serviceNames: string[] = [];
+    if (serviceList.items !== undefined) {
+      serviceNames = serviceList.items!.map(
+        (service: run_v1.Schema$Service) => service.metadata!.name as string,
+      );
+    }
+    return serviceNames;
+  }
+
+  /**
+   * Deploy a Cloud Run service.
+   *
+   * @param service Service object
+   * @returns Service object
+   */
+  async deploy(service: Service): Promise<run_v1.Schema$Service> {
+    const authClient = await this.getAuthClient();
+    const serviceNames = await this.listServices();
+    let serviceResponse: GaxiosResponse<run_v1.Schema$Service>;
+    if (serviceNames!.includes(service.name)) {
+      core.info('Creating a service revision...');
+      // Replace service
+      const createServiceRequest: run_v1.Params$Resource$Projects$Locations$Services$Replaceservice = {
+        name: this.getResource(service),
+        auth: authClient,
+        requestBody: service.request,
+      };
+      serviceResponse = await this.run.projects.locations.services.replaceService(
+        createServiceRequest,
+        this.methodOptions,
+      );
+    } else {
+      core.info('Creating a new service...');
+      // Create service
+      const createServiceRequest: run_v1.Params$Resource$Projects$Locations$Services$Create = {
+        parent: this.parent,
+        auth: authClient,
+        requestBody: service.request,
+      };
+      serviceResponse = await this.run.projects.locations.services.create(
+        createServiceRequest,
+        this.methodOptions,
+      );
+    }
+    core.info(`Service ${service.name} has been successfully deployed.`);
+    return serviceResponse.data;
+  }
+
+  /**
+   * Deletes a Cloud Run service.
+   *
+   * @param service Service object
+   */
+  async delete(service: Service): Promise<void> {
```

Comment:
Are we supporting a delete feature or is this just for cleaning up the test?
https://github.com/GoogleCloudPlatform/github-actions/blob/828bf4e608a7cc05a79e539b6bb9b5354bdce7eb/deploy-cloudrun/tests/unit/cloudRun.test.ts#L61

Reply:
Just for removing test resources